src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class encapsulates a list of Tasks.
+ *
+ * @author Kleon Ang
+ */
+public class TaskList extends ArrayList<Task> {
```

Comment:
interesting way to go bout w TaskList, perhaps a composition might be more suitable in case you want to add other details inside TaskList like timeAccessed juz to name a random example. With inheritance, its somewhat more limiting if you follow the logic that TaskList is an ArrayList (imo)

Reply:
I think it works to add local fields to TaskList as an inherited ArrayList if we require additional variables, plus inheritance allows me to use the predefined iterator as well as `ArrayList::add` and `ArrayList::remove` functions for testing too 😅. Nevertheless, I'll keep your suggestion in mind in case this change is necessary in the future. Thanks!